teps/0122-complete-build-instructions-and-parameters.md

```diff
@@ -0,0 +1,241 @@
+- [Configuration feature flags](#configuration-feature-flags)
+- [Materials](#materials)
+
+This will allow us to create a provenance for a task run as [shown](#provenance-for-executed-taskrun).
+
+### Task Specification
+The table below provides the name and description of the API fields that the Task requires. In addition, it also lists if that information is required, not required, or provided by the provenance. If it is required, it also shows where the information should be provided. Looking at the table below and required [step](#step-specification) information, there are only a handful of fields that are not required for reproducibility. Instead of cherry-picking most of the fields, including the entire taskSpec in the buildConfig section of the provenance is likely the best way to go forward. Additionally, if we update the spec in the future, then chains will not have to be updated correspondingly. The work described in [issue](https://github.com/tektoncd/chains/issues/597) is meant to handle this.
+
+
+| | **Metadata**      | | |
+| ------ | ------------- | ----- | ----- |
+| **Field Name**    | **Description** | **Not Required / Required / Provided by provenance** | **Insert in provenance** |
+| name          | Name of the task | Not Required (if the spec is already in the provenance) | |
+| Spec          | | | |
+| resources     | resources used by steps | Not Required (Deprecated) |  |
+| description   | description of the task | Not Required |  |
+| params        | parameters required by task | Provided |  |
+| results       | results produced by the task | Required(Atleast name, type and  properties (in case of object)) | buildConfig |
+| volumes       | volume mounted on the container | Required | buildConfig |
+| workspaces    | workspace bindings used by the task | Required | buildConfig |
+| steps         | Steps performed by the task | Provided |  |
+| sidecars      | Sidecars running alongside tasks | Required | buildConfig |
+| step template | Specifies a container configuration that will be used as the starting point for all of the Steps in your Task | Required | buildConfig |
+
+### Step Specification
+
+The table below provides the name and description of the API fields that the Step requires. In addition, it also lists if that information is required, not required, or provided by the provenance. If it is required, it also shows where the information should be provided. **Deprecated fields have not been included in the table below.**  Like in [Task Specification](#task-specification), almost all fields are either required or provided in the provenance. Therefore for the same reasons, we should include the entire spec in the generated provenance.
+
+
+| | **Metadata**      | | |
+| ------ | ------------- | ----- | ----- |
+| **Field Name**    | **Description** | **Not Required / Required / Provided by provenance** | **Insert in provenance** |
+| name             | Name of the step | Required | buildConfig |
```

Comment:
The step name is not mandatory in Tekton - even though we discourage nameless steps.
Would it be a problem in this case?

Reply:
Not a problem. I will fix it.